src/app/modules/zxing-scanner/browser-code-reader.ts

```diff
@@ -86,15 +90,19 @@ export class BrowserCodeReader {
      * @param deviceId The device's to be used Id
      * @param videoElement A new video element
      */
-    public decodeFromInputVideoDevice(callbackFn: (result: Result) => any, deviceId?: string, videoElement?: HTMLVideoElement): void {
+    public decodeFromInputVideoDevice(callbackFn?: (result: Result) => any, deviceId?: string, videoElement?: HTMLVideoElement): void {
+
+        if (deviceId !== undefined) {
+            this.deviceId = deviceId;
```

Comment:
We have to add a setter for this guy later, there's really no need for this now, just later.